### PR TITLE
multiple devices error

### PR DESCRIPTION
**Pull request by gaospecial**
When set `dev = c("png","pdf")`, it will cause a error:

```
Error in suffix != "" && TRUE : 'length = 2' in coercion to 'logical(1)'
```

This PR fixed this problem.

### CODE REVIEW

File: R/utils.R [hunk 1]
```diff
@@ -511,7 +511,7 @@ pandoc_fragment = function(text, to = pandoc_to(), from = pandoc_from()) {
 #' @examples fig_path('.pdf', options = list(fig.path='figure/abc-', label='first-plot'))
 #' fig_path('.png', list(fig.path='foo-', label='bar'), 1:10)
 fig_path = function(suffix = '', options = opts_current$get(), number) {
-  if (suffix != '' && !grepl('[.]', suffix)) suffix = paste0('.', suffix)
+  if (all(suffix != '') && all(!grepl('[.]', suffix))) suffix = paste0('.', suffix)
```
Comment (yihui) [inline R/utils.R:514]: I think this can be simplified to

```suggestion
  suffix = sub('^([^.])', '.\\1', suffix)
```

**Review by yihui — APPROVED** [review]
This sounds reasonable, but I don't know how to reproduce the problem. I tested a simple code chunk and didn't see the error:

````
```{r, dev=c('png', 'pdf')}
plot(1)
```
````

Thanks!

**Comment by gaospecial** [comment]
I work with a bookdown project, and my session info is as follows:

```
## R version 4.3.2 (2023-10-31)
## Platform: aarch64-apple-darwin20 (64-bit)
## Running under: macOS 15.0.1
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
## LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## time zone: Asia/Singapore
## tzcode source: internal
## 
## attached base packages:
## [1] grDevices datasets  grid      graphics  stats4    stats     utils    
## [8] methods   base     
## 
## other attached packages:
##  [1] ggraph_2.2.1                           
##  [2] igraph_2.0.3                           
##  [3] org.Hs.eg.db_3.18.0                    
##  [4] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
##  [5] Biostrings_2.68.1                      
##  [6] XVector_0.40.0                         
##  [7] phyloseq_1.44.0                        
##  [8] gtable_0.3.5                           
##  [9] ggpubr_0.6.0                           
## [10] org.EcoliSakai.eg.db_0.1               
## [11] apeglm_1.22.1                          
## [12] createKEGGdb_0.0.3                     
## [13] KEGG.db_1.0                            
## [14] UpSetR_1.4.0                           
## [15] sva_3.48.0                             
## [16] BiocParallel_1.34.2                    
## [17] genefilter_1.82.1                      
## [18] mgcv_1.9-0                             
## [19] nlme_3.1-163                           
## [20] pheatmap_1.0.12                        
## [21] RColorBrewer_1.1-3                     
## [22] ggtext_0.1.2                           
## [23] cailab.utils_0.3.2                     
## [24] ChIPseeker_1.41.3                      
## [25] ggVennDiagram_1.5.0                    
## [26] rtracklayer_1.60.1                     
## [27] gdtools_0.3.5                          
## [28] ggtree_3.8.2                           
## [29] GenomicFeatures_1.52.2                 
## [30] AnnotationDbi_1.64.1                   
## [31] genbankr_1.27.0                        
## [32] conflicted_1.2.0                       
## [33] enrichplot_1.25.3                      
## [34] clusterProfiler_4.10.0                 
## [35] DESeq2_1.42.0                          
## [36] SummarizedExperiment_1.30.2            
## [37] Biobase_2.60.0                         
## [38] MatrixGenerics_1.12.3                  
## [39] matrixStats_1.4.1                      
## [40] tidyverse_2.0.0                        
## [41] tidyr_1.3.1                            
## [42] tibble_3.2.1                           
## [43] stringr_1.5.1                          
## [44] readr_2.1.4                            
## [45] purrr_1.0.2                            
## [46] lubridate_1.9.3                        
## [47] ggplot2_3.5.1                          
## [48] forcats_1.0.0                          
## [49] dplyr_1.1.4                            
## [50] Gviz_1.44.2                            
## [51] GenomicRanges_1.52.1                   
## [52] GenomeInfoDb_1.36.4                    
## [53] IRanges_2.34.1                         
## [54] S4Vectors_0.38.2                       
## [55] BiocGenerics_0.46.0          
```

**Comment by yihui** [comment]
Even if it's your problem, I think the code change in this PR will be an improvement (the operation on `suffix` has been vectorized), so I'll merge it anyway.

**Comment by gaospecial** [comment]
OK. I agree. ^_^

**Pull request merged**